program/src/matching.rs

```diff
@@ -1181,12 +1185,13 @@ impl<'a> Book<'a> {
         //
         // Any changes to matching asks are collected in ask_changes
         // and then applied after this loop.
-        let mut rem_quantity = quantity; // base lots (aka contracts)
+        let mut rem_base_quantity = max_base_quantity; // base lots (aka contracts)
+        let mut rem_quote_quantity = max_quote_quantity;
         let mut ask_changes: Vec<(NodeHandle, i64)> = vec![];
         let mut ask_deletes: Vec<i128> = vec![];
         let mut number_of_dropped_expired_orders = 0;
         for (best_ask_h, best_ask) in self.asks.iter_all_including_invalid() {
-            if rem_quantity <= 0 {
+            if rem_base_quantity <= 0 || rem_quote_quantity <= 0 {
```

Comment:
This exit condition doesn't work, since `rem_quote_quantity` can stay >0 due to rounding effects.